recipe/meta.yaml

```diff
@@ -376,7 +376,7 @@ outputs:
         # tools/ is needed to optimise test run
         # as of pytorch=2.0.0, there is a bug when trying to run tests without the tools
         - tools
-        - .ci/pytorch/smoke_test/smoke_test.py
+        #- .ci/pytorch/smoke_test/smoke_test.py
```

Comment:
Looks like https://github.com/pytorch/pytorch/commit/f57606ab850bd304cfb4a16f8ac62cd66fe23bbe landed only for 2.6.0, which is around the corner, so this still makes sense to keep at least.

Reply:
I was going to ask if you wanted this on the 2.6.0 PR; here, it blocks the rest of the testing because this is still on 2.5.1

Reply:
logically it makes more sense to add this in the 2.6 PR, but I don't really mind if it lands (though deactivated) in this PR already

Reply:
As an aside, for further development flow optimization, we could run just the smoke tests on non-release PRs.

Reply:
What kind of runtime are we expecting the smoketest to take? If it's <30min, it's fine to run always IMO.

Reply:
it's pretty brief, much briefer than the suite of tests. but what I mean is: for non-release PRs, only run the smoke tests, don't run the rest of the tests. Then for releases, run all the tests.

Reply:
What's a "non-release PR"? One for a release candidate? Or what do you mean there?

Reply:
I mean a PR where features are added but upon merge, a new conda package won't be uploaded to conda-forge. "release" from the point of view of the conda package release.

Reply:
OK, that's what confused me, because we never release anything based on what's built in a PR, it needs to be merged anyway. And I don't agree to not run the full tests in PRs, that just means we get negative surprises _after_ merging (and the builds that passed ended up being published anyway, creating an inconsistent situation across build combination).

Reply:
> and the builds that passed ended up being published anyway, creating an inconsistent situation across build combination

I think there's something I don't understand about the workflow here too. are the builds for all merged PRs published? I think the answer is yes.

Reply:
Yes, every merge to `main` will publish builds. This is a core constraint in our setup, we don't have a switch that says "merge to main but don't publish" (well except from manually adding a `skip: true` in the recipe).

Reply:
Correction: such a switch does exist:
https://github.com/conda-forge/pytorch-cpu-feedstock/blob/4ff92b330141d5d496afdc2d83ca0f32f6adf03d/.scripts/build_steps.sh#L91-L93
but I've never seen `UPLOAD_PACKAGES` used.